op-challenger/game/factory.go

```diff
@@ -56,6 +74,7 @@ func (l *gameLoader) FetchAllGamesAtBlock(ctx context.Context, earliestTimestamp
 		return nil, fmt.Errorf("failed to fetch game count: %w", err)
 	}
 
+	l.m.ClearTrackedGames()
```

Comment:
We need to do the update more atomically than this, otherwise the metrics may get sampled in between this clearing and the games being recounted which will mess with the stats.

I suspect this loader isn't the right place to try and add this metric given it doesn't know the game status either.  The scheduler coordinator might be a better place. It tracks the `gameState` already and while that currently is just tracking status as resolved or not, it would actually be quite reasonable to move that to tracking the actual status. It would mean the `GamePlayer.ProgressGame` returns the status instead of just a `bool` but I think that's probably clearer anyway.